ddtrace/span.py

```diff
@@ -207,9 +207,9 @@ def finish(self, finish_time=None):
             self.duration_ns = ft - (self.start_ns or ft)
 
         if self._context:
-            trace, sampled = self._context.close_span(self)
-            if self.tracer and trace and sampled:
-                self.tracer.write(trace)
+            self._context.close_span(self)
+        for cb in self._on_finish_callbacks:
+            cb(self)
```

Comment:
do we want to handle errors there?

Reply:
Hmm it's tricky. On one hand I think it should be the responsibility of the owner of a span to handle any exceptions that might be raised by the callbacks but on the other the callbacks happen at a later point and may not even be handle-able when using span as a context manager. 🤔 

I'm personally leaning towards the former rather than the latter. WDYT?